Frameworks/UIKit/UIButton.mm

```diff
@@ -404,19 +415,34 @@ - (CGRect)imageRectForContentRect:(CGRect)contentRect {
  @Status Interoperable
 */
 - (CGRect)titleRectForContentRect:(CGRect)contentRect {
-    // TODO: Should we be asking our label for its intrinsicContentSize?
-    // We need to round the font size to the nearest pixel value to avoid rounding errors when used in conjunction with the
-    // frame values that are rounded by UIView.
-    CGSize titleSize = doPixelRound([self.currentTitle sizeWithFont:self.font]);
-    CGSize totalSize = titleSize;
-    // TODO  #1365 :: Currently cannot assume getting size from nil will return CGSizeZero
-    CGSize imageSize = self.currentImage ? [self.currentImage size] : CGSizeZero;
-    CGRect insetsRect = UIEdgeInsetsInsetRect(contentRect, self.titleEdgeInsets);
+    /////////////////////////////////////////////////////////////////////
+    // Note: #1365 All size calculations here must check for nil, 
+    // as we cannot assume getting a size from nil will return CGSizeZero
+    /////////////////////////////////////////////////////////////////////
 
-    if ([self currentTitle].length == 0) {
+    // Grab the size of the current title
+    // TODO: Should we be asking our label for this?
+    NSString* currentTitle = self.currentTitle;
+    CGSize titleSize = CGSizeZero;
+    if (currentTitle && currentTitle.length > 0) {
```

Comment:
i know, but if we don't leave the explicit nil check, someone may forget next time they come through here?

---
In reply to: [103035907](https://github.com/Microsoft/WinObjC/pull/2094#discussion_r103035907) [](ancestors = 103035907)